docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+
+## Temporal.Date internal slots
+
+Temporal.Date currently has three internal slots: year, month, and day. (An "internal slot" refers to actual data, as opposed to "properties", which could be computed.)  In this proposal, three more internal slots would be added:
+
+1. calendar
+2. isLeapMonth
+3. era
+
+It is believed that a date in any known calendar can be described using the five-tuple (*year*, *month*, *day*, *isLeapMonth*, *era*).  It is possible that *isLeapMonth* can be merged into *month* by expanding the set of numbers that *month* can represent.
+
+The *calendar* slot contains an object implementing the Temporal.Calendar interface, described below.
+
+## Temporal.Calendar interface
+
+The new Temporal.Calendar interface is a mechanism to allow arbitrary calendar systems to be implemented on top of Temporal.  ***Most users will not encounter the Temporal.Calendar interface directly***, unless they are building or using a non-built-in calendar system.
+
+### Methods on the Temporal.Calendar interface
+
+All of the following methods return new Temporal.Date objects.  All properties of `Temporal.Calendar` are Symbols.
+
+```javascript
+class MyCalendar {
+
+	///////////////////
+	//  To/From ISO  //
+	///////////////////
+
+	/** Returns the projection of self in the ISO calendar */
+	[Temporal.Calendar.toISO](
+		self: Temporal.Date
+	) : Temporal.Date;
+
+	/** Returns the projection of isoDate in the custom calendar */
+	[Temporal.Calendar.fromISO](
+		isoDate: Temporal.Date
+	) : Temporal.Date;
+
+	/** A string identifier for this calendar */
+	[Temporal.Calendar.id] : string;
+
+	//////////////////
+	//  Arithmetic  //
+	//////////////////
+
+	/** Returns self plus duration according to the calendar rules. */
+	[Temporal.Calendar.plus](
+		self: Temporal.Date,
+		duration: Temporal.Duration,
+		options: /* options bag */
+	) : Temporal.Date;
+
+	/** Returns self minus duration according to the calendar rules. */
+	[Temporal.Calendar.minus](
+		self: Temporal.Date,
+		duration: Temporal.Duration,
+		options: /* options bag */
+	) : Temporal.Date;
+
+	/** Returns self minus other, which are dates in the same calendar. */
+	[Temporal.Calendar.difference](
+		self: Temporal.Date,
+		other: Temporal.Date,
+		options: /* options bag */
+	) : Temporal.Duration;
+
+	////////////////////////////////////
+	//  Accessors:                    //
+	//  Semantics defined in date.md  //
+	////////////////////////////////////
+
+	[Temporal.Calendar.dayOfWeek](
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.weekOfYear](
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.daysInMonth](
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.daysInYear](
+		self: Temporal.Date
+	) : number;
+
+	[Temporal.Calendar.leapYear](
+		self: Temporal.Date
+	) : number;
+}
+```
+
+An instance of `MyCalendar` would be immutable.
```

Comment:
What enforces this?

Reply:
I meant "stateless" rather than "immutable"; I rewrote the sentence.

Reply:
What enforces that, then?

Reply:
I rewrote the paragraph again:

> An instance of `MyCalendar` is *expected* to have stateless behavior; i.e., calling a method with the same arguments should return the same result each time.  There would be no mechanism for enforcing that user-land calendars are stateless; the calendar author should test this expectation on their own in order to prevent unexpected behavior such as the lack of round-tripping.